cmd/acraserver/listener.go

```diff
@@ -15,35 +15,73 @@ package main
 
 import (
 	"net"
-
 	"github.com/cossacklabs/acra/network"
 	log "github.com/sirupsen/logrus"
-
 	"github.com/cossacklabs/acra/decryptor/base"
 	pg "github.com/cossacklabs/acra/decryptor/postgresql"
 	"github.com/cossacklabs/acra/keystore"
 	"github.com/cossacklabs/acra/zone"
+	"os"
+	"time"
+	"errors"
+	"syscall"
+	url_ "net/url"
 )
 
 type SServer struct {
-	config     *Config
-	keystorage keystore.KeyStore
-	listeners  []net.Listener
+	config       *Config
+	keystorage   keystore.KeyStore
+	listenerACRA net.Listener
+	listenerAPI  net.Listener
+	fddACRA      uintptr
+	fdAPI        uintptr
+	cmACRA       *network.ConnectionManager
+	cmAPI        *network.ConnectionManager
+	listeners    []net.Listener
 }
 
 func NewServer(config *Config, keystorage keystore.KeyStore) (server *SServer, err error) {
-	return &SServer{config: config, keystorage: keystorage}, nil
+	return &SServer{
+		config:     config,
+		keystorage: keystorage,
+		cmACRA:     network.NewConnectionManager(),
+		cmAPI:      network.NewConnectionManager(),
+	}, nil
 }
 
 // Close all listeners and return first error
-func (server *SServer) Close() error {
+func (server *SServer) Close() {
+	log.Debugln("Closing listeners")
 	var err error
 	for _, listener := range server.listeners {
-		if err_ := listener.Close(); err_ != nil && err == nil {
-			err = err_
+		switch listener.(type) {
+		case *net.TCPListener:
+			listener.(*net.TCPListener).Close()
+		case *net.UnixListener:
+			err = listener.(*net.UnixListener).Close()
+			if err != nil {
+				log.WithError(err).Infoln("UnixListener.Close()")
+				continue
+			}
+			// TODO: find better way to remove unixsocket file
+			f, err1 :=listener.(*net.UnixListener).File()
+			_ = f
+			if err1 != nil {
+				log.WithError(err).Warningln("UnixListener.Close File()")
```

Comment:
why not log.WithError(err**1**)? We will not know anything about error message from `err1`.

Reply:
done